api/src/pcapi/utils/chunks.py

```diff
@@ -24,3 +24,19 @@ def get_chunks(input_data: typing.Iterable, chunk_size: int) -> typing.Generator
 
         if len(chunk) < chunk_size:
             break
+
+
+def split_sequence(input_data: typing.Sequence[T], max_length: int) -> list[typing.Sequence[T]]:
+    """
+    Split sequence into smaller parts of max `max_length` items each.
+    eg. split_sequence([1, 2, 3, 4, 5], 3) -> [[1, 2, 3], [4, 5]]
+
+    Note:
+        This function is designed to work well for creating a small
+        number of splits. Avoid calling it if you might need to create
+        lots (hundreds or more) of smaller parts.
+    """
+    if len(input_data) <= max_length:
+        return [input_data]
+
+    return [input_data[:max_length], *split_sequence(input_data[max_length:], max_length)]
```

Comment:
```suggestion
def split_sequence(input_data: typing.Sequence[T], max_length: int, acc = []) -> list[typing.Sequence[T]]:
    """
    Split sequence into smaller parts of max `max_length` items each.
    eg. split_sequence([1, 2, 3, 4, 5], 3) -> [[1, 2, 3], [4, 5]]
    """
    if len(input_data) <= max_length:
        acc.append(input_data)
        return acc

    acc.append(input_data[:max_length])
    return split_sequence(input_data[max_length:], max_length, acc)
```

Pour faire une tail recursion ? Si tu fonctionne avec des très grosses listes et des petits chunks ça pourrait poser problème, à voir si c'est utile

Reply:
`split_sequence` est supprimée, `get_chunks` faisait déjà la même chose.